### ISSUE TITLE
Unused characters in Dutch grid

### ISSUE THREAD

**Issue by lgommans** [lexica/lexica#107]
These are the letters used in the Dutch grid:

https://github.com/lexica/lexica/blob/master/app/src/main/res/raw/letters_nl.txt

They are a little weird.

To take an easy example, the î is not a letter we use, but unlike the English, we usually leave accents when we loan words from other languages (the most commonly known example in both English and Dutch might be [crème fraîche](https://en.wikipedia.org/wiki/Cr%C3%A8me_fra%C3%AEche)), but that does not mean that the circumflex on the i (î) is a known letter to Dutch people. You don't have an î in the English grid either!

We use a diaeresis (dots above it) to make something a separate sound, so the word idee (idea) in plural is ideeën (ideas). The pronunciation between Dutch and English in singular is almost the same, save for the "a" at the end in English. In plural, we just add "en" in the end, but because it's preceded by an "e" and sequential "e"s are a single sound, we need to make clear that it's a new sound. So the e with diaeresis (ë) is not a different letter, it's just separating one "e" sound from another. Dictionary search engines like woorden.org ('words'.org) also happily include the diaeresis for you if you omit them when looking up a word: it's a detail used in writing, we omit them from laziness without loss of understandableness.

Note that this is very different from the German usage, where an "ä" is unequal to "a" in sound, and the word "nass" is definitely considered unequal from "näss".

We do have some sound modifiers, like for the word café (which means, unsurprisingly, cafe), but we also use accents for emphasis ("geen paarden!" -> "géén paarden!" could be used to mean the same as "no horses!" -> "no horses!"). This shows that we consider the letter mostly equal with only a different emphasis.

For loanwords, I guess the è, ç, and û could appear in Dutch print, but I bet the number of Dutch people who can find those keys on a keyboard is roughly equal to the number of words with such characters in the [Dutch dictionary](https://github.com/lexica/lexica/tree/master/assets/dictionaries). While accents and diaeresis are taught in school, the remaining characters are really not part of the language at all (and we only use diaeresis commonly for the i and the e). Only 1 promille (138 words) in the Dutch Lexica dictionary contain any of è ñ ç û ô à ä î. The x in English occurs in 2% of words (1122 in the en_GB dictionary), so that one is super useful by comparison ;)

(There is also the letter IJ/Y: my grandma sometimes replaces ij with y to save characters in SMS messages, in old Dutch you might see the y used instead of ij, and some online dictionaries will redirect you from a word like "ys" to "ijs" (means "ice"), but it's very old fashioned. As a Dutchman, I think it's fine to ignore.)

---

So what should we do regarding those semi-letters? I don't know if this discussion was held already (I couldn't find it in past Github issues), and I am not a linguist, but I would say that the grid should be only A-Z. The expected behaviour would be that swiping "ideeen" matches "ideeën" and swiping "cafe" matches "café".

The easy solution is to just replace the letters by their diacriticless bases in the dictionary, but here's the catch: I do think Dutch players would expect ideeën to show up spelled correctly. I think the game logic will need to include a few lines for Dutch to check the dictionary in diacriticless manner and then saves the word to the matched list correctly.

What do you think?

**Comment by d-mal** [comment]
Yes, as you suspect, there is an issue about this already: #71 is probably the main one, #126 is a duplicate.

**Comment by lgommans** [comment]
Thanks for the reply d-mal! Je aussie ne parle pas français, so I didn't know that the French behaviour would be the same as Dutch. Perhaps I skipped the thread because of that, or perhaps I just searched for Dutch, got no results, and figured I should open a new ticket. Either way, reading that thread now, it seems all languages are chiming in there, so let's do that for Dutch as well and close this as a duplicate :). We can reopen or start a new ticket if Dutch ever becomes a corner case, though it sounds like French and Spanish are basically the same.
